File: WageCalculator/ViewModels/DailyWage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using WageCalculator.Entities;
using WageCalculator.Helpers;

namespace WageCalculator.ViewModels
{
    /// <summary>
    /// Visualize the daily wage calculations
    /// </summary>
    public class DailyWage
    {
        public decimal TotalWage => NormalHoursWage + EveningCompensation + OvertimeCompensation;
        public decimal NormalHoursWage { get; set; }
        public decimal EveningCompensation { get; set; }
        public decimal OvertimeCompensation { get; set; }
        public decimal WorkingHours { get; set; }
        public decimal EveningHours { get; set; }
        public decimal OvertimeHours { get; set; }
        public WorkingDay WorkingDay { get; set; }
        public WagePricing WagePricing { get; set; }
    }
}

File: WageCalculator/Scripts/src/dailyWage.js
import React from "react";
import moment from "moment";
import WorkingShift from "./workingShift";

export default React.createClass({
    render: function() {
        var workingShiftKey = 0;
        var workingShifts = this.props.dailyWage.WorkingDay.WorkingShifts.map(function(workingShift) {
            workingShiftKey++;
            return (
               <WorkingShift key={workingShiftKey} workingShift={workingShift} />
        );
      });
        return (
         <tr>
            <td>
                    {moment(this.props.dailyWage.WorkingDay.Date).format('MMMM Do YYYY')}
            </td>
            <td>
                    {workingShifts}
            </td>
               <td>
                    {this.props.dailyWage.TotalWage} $<br/>
                    {this.props.dailyWage.WorkingHours} h
                </td>
                 <td>
                     {this.props.dailyWage.EveningCompensation} $<br/>
                    {this.props.dailyWage.EveningHours} h
                </td>
                <td>
                    {this.props.dailyWage.OvertimeCompensation} $<br/>
                    {this.props.dailyWage.OvertimeHours} h
                </td>
            </tr>
        );
    }
});
